File: solutions/csharp/robot-simulator/1/RobotSimulator.cs
using System;

public enum Direction
{
    North,
    East,
    South,
    West
}

public class RobotSimulator
{
    private Direction _direction { get; set; }
    private int _x { get; set; }
    private int _y { get; set; }
    
    public RobotSimulator(Direction direction, int x, int y)
    {
        _direction = direction;
        _x = x;
        _y = y;
    }

    public Direction Direction
    {
        get
        {
            return _direction;
        }
    }

    public int X
    {
        get
        {
            return _x;
        }
    }

    public int Y
    {
        get
        {
            return _y;
        }
    }

    public void Move(string instructions)
    {
        foreach (char instruction in instructions)
        {
            if (instruction == 'L')
            {
                _direction = (Direction)(((int)_direction + 3) % 4);
            }
            else if (instruction == 'R')
            {
                _direction = (Direction)(((int)_direction + 1) % 4);
            }
            else if (instruction == 'A')
            {
                switch (_direction)
                {
                    case Direction.North:
                        _y++;
                        break;
                    case Direction.East:
                        _x++;
                        break;
                    case Direction.South:
                        _y--;
                        break;
                    case Direction.West:
                        _x--;
                        break;
                }
            }
            else
            {
                throw new ArgumentException("Invalid instruction.");
            }
        }
    }
}

File: solutions/python/robot-simulator/1/robot_simulator.py
# Globals for the directions
EAST = 'east'
NORTH = 'north'
WEST = 'west'
SOUTH = 'south'


class Robot:
    def __init__(self, direction=NORTH, x_pos=0, y_pos=0):
        self.direction = direction
        self.coordinates = (x_pos, y_pos)

    def turn_right(self):
        if self.direction == NORTH:
            self.direction = EAST
        elif self.direction == EAST:
            self.direction = SOUTH
        elif self.direction == SOUTH:
            self.direction = WEST
        elif self.direction == WEST:
            self.direction = NORTH

    def turn_left(self):
        if self.direction == NORTH:
            self.direction = WEST
        elif self.direction == WEST:
            self.direction = SOUTH
        elif self.direction == SOUTH:
            self.direction = EAST
        elif self.direction == EAST:
            self.direction = NORTH

    def advance(self):
        if self.direction == NORTH:
            self.coordinates = (self.coordinates[0], self.coordinates[1] + 1)
        elif self.direction == EAST:
            self.coordinates = (self.coordinates[0] + 1, self.coordinates[1])
        elif self.direction == SOUTH:
            self.coordinates = (self.coordinates[0], self.coordinates[1] - 1)
        elif self.direction == WEST:
            self.coordinates = (self.coordinates[0] - 1, self.coordinates[1])

    def move(self, instructions):
        for instruction in instructions:
            if instruction == 'R':
                self.turn_right()
            elif instruction == 'L':
                self.turn_left()
            elif instruction == 'A':
                self.advance()

    def get_coordinates(self):
        return self.coordinates

    def get_bearing(self):
        return self.direction
